polaris-migrator/src/migrations/replace-sass-lengths/replace-sass-lengths.ts

```diff
@@ -0,0 +1,187 @@
+ */
+const ExitAndStopTraversing = false;
+
+interface PluginOptions extends Options, NamespaceOptions {}
+
+const plugin = (options: PluginOptions = {}): Plugin => {
+  const remFunction = namespace('rem', options);
+  const isRemFunction = createIsSassFunction(remFunction);
+
+  return {
+    postcssPlugin: 'replace-sass-lengths',
+    Declaration(decl) {
+      // @ts-expect-error - Skip if processed so we don't process it again
+      if (decl[processed]) return;
+
+      const prop = decl.prop;
+      const parsedValue = valueParser(decl.value);
+
+      if (!isTargetProp(prop)) return;
+
+      parsedValue.walk((node) => {
+        if (isRemFunction(node)) {
+          const argDimension = valueParser.unit(node.nodes[0]?.value ?? '');
+
+          if (
+            argDimension &&
+            supportedDimensionUnits.includes(argDimension.unit)
+          ) {
+            const argInPx = toPx(`${argDimension.number}${argDimension.unit}`);
+
+            if (!isSpacingTokenValue(argInPx)) return ExitAndStopTraversing;
+
+            const spacingToken = spacingTokensMap[argInPx];
+
+            node.value = 'var';
+            node.nodes = [
+              {
+                type: 'word',
+                value: spacingToken,
+                sourceIndex: node.nodes[0]?.sourceIndex ?? 0,
+                sourceEndIndex: spacingToken.length,
+              },
+              ...node.nodes.slice(1),
+            ];
+          }
+
+          return ExitAndStopTraversing;
+        } else if (node.type === 'word') {
+          const dimension = valueParser.unit(node.value);
+
+          if (dimension && supportedDimensionUnits.includes(dimension.unit)) {
+            const dimensionInPx = toPx(`${dimension.number}${dimension.unit}`);
+
+            if (!isSpacingTokenValue(dimensionInPx)) return;
+
+            node.value = `var(${spacingTokensMap[dimensionInPx]})`;
+          }
+        }
+      });
+
+      if (hasCalculation(parsedValue)) {
+        // Insert comment if the declaration value contains calculations
+        decl.before(postcss.comment({text: POLARIS_MIGRATOR_COMMENT}));
+        decl.before(
+          postcss.comment({text: `${decl.prop}: ${parsedValue.toString()};`}),
+        );
+      } else {
+        decl.value = parsedValue.toString();
+      }
+
+      // @ts-expect-error - Mark the declaration as processed
+      decl[processed] = true;
+    },
+  };
+};
+
+export default function replaceSassLengths(
+  fileInfo: FileInfo,
+  _: API,
+  options: Options,
+) {
+  return postcss(plugin(options)).process(fileInfo.source, {
+    syntax: require('postcss-scss'),
+  }).css;
+}
+
+function hasCalculation(parsedValue: valueParser.ParsedValue): boolean {
+  let hasCalc = false;
+
+  parsedValue.walk((node) => {
+    if (isNumericOperator(node)) hasCalc = true;
+  });
+
+  return hasCalc;
+}
+
+function isNumericOperator(node: Node): boolean {
```

Comment:
Should we move this function to the `sass.ts` utilities? Then we can also reuse it in the `replace-sass-spacing` migrations 👏

Reply:
Yes! Addressed in https://github.com/Shopify/polaris/pull/7264/commits/f6e47fa81dfe256b7fc4ec787a86e26381735a92